x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/formatter/TextFormat.java

```diff
@@ -313,9 +320,16 @@ public String contentType(RestRequest request) {
     }
 
     // utility method for consuming a row.
-    <F> void row(Writer writer, Iterator<F> row, Function<F, String> toString, Character delimiter) throws IOException {
+    <F> void row(Writer writer, Iterator<F> row, Function<F, String> toString, Character delimiter, boolean[] nullColumns)
+        throws IOException {
         boolean firstColumn = true;
+        int i = -1;
         while (row.hasNext()) {
+            i++;
+            if (nullColumns != null && nullColumns[i]) {
```

Comment:
It'd make it just an array of true rather than nullable. A little easier to think about and not much extra memory used.

Reply:
Thanks @nik9000. I suppose you mean an array of false when there is no need to drop null columns.

Reply:
Er, yeah. Right. Don't drop. Maybe we should name it `dropColumns`? That'd be a little cleared about what it's *for* in this context. Super minor thing.

Reply:
Yes, I agree. It can be confusing otherwise.